### DISCUSSION TITLE
WIP: Microsoft.ML.TorchSharp

### DISCUSSION THREAD

**Discussion by GeorgeS2019** [xamarin/TorchSharp#651] [Ideas]
Starting[ Jun 2022](https://github.com/dotnet/machinelearning/projects/27#card-75721623), there is now an official [Microsoft.ML.TorchSharp](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.torchsharp) namespace

Starting this discussion here to lobby exchange of views on this topic :-)

**Comment by GeorgeS2019** [comment]
Based on the implemented [Microsoft.ML.TorchSharp text classification example](https://github.com/dotnet/machinelearning/blob/main/test/Microsoft.ML.Tests/TextClassificationTests.cs)

This is one of the SIX categories of TorchText use cases

[torchtext.datasets](https://pytorch.org/text/stable/datasets.html)

- [Text Classification](https://pytorch.org/text/stable/datasets.html#text-classification)
- [Language Modeling](https://pytorch.org/text/stable/datasets.html#language-modeling)
- [Machine Translation](https://pytorch.org/text/stable/datasets.html#machine-translation)
- [Sequence Tagging](https://pytorch.org/text/stable/datasets.html#sequence-tagging)
- [Question Answer](https://pytorch.org/text/stable/datasets.html#question-answer)
- [Unsupervised Learning](https://pytorch.org/text/stable/datasets.html#unsupervised-learning)

Questions: 
- Is there a need for documentation to improve the synergy of community contributions in ML.NET and TorchSharp to accelerate the coverage of deep learning use cases for .NET?

For example, what have been done in TorchSharp Vision and Audio within the TorchSharp community could speed up the coverage of deep learning for vision and audio in ML.NET.  Likewise for deep learning for NLP done in TorchSharp and NL.NET.

**Comment by GeorgeS2019** [comment]
Within **Question Answer** deep learning for NLP use case

There is now a TorchSharp [example ](https://github.com/dotnet/TorchSharpExamples/pull/19)based on [SQuAD 2.0](https://pytorch.org/text/stable/datasets.html#id34) dataset

@QingtaoLi1

Would you be interested to rewrite your contribution to ways that will be more aligned with the TorchText for TorchSharp framework as well as Microsoft.ML.TorchSharp for ML.NET?

**Comment by GeorgeS2019** [comment]
## [Deep Voice/Audio in ML.NET](https://github.com/dotnet/machinelearning/issues/6069)

FYI: @kaiidams 

@papyr
[Voice and Audio request](https://github.com/dotnet/machinelearning/issues/6069) for Microsoft.ML.TorchSharp could now explore the [recent work done on TorchAudio for TorchSharp](https://github.com/dotnet/TorchSharp/pull/649)

@Neeraj-Kashyap-Harman
[Can speech recognition be done in ML.NET?](https://github.com/dotnet/machinelearning/issues/2732)

@lukaszadamus
[Audiogram in ML.NET](https://github.com/dotnet/machinelearning/issues/2134#issuecomment-865702462)

**Comment by NiklasGustafsson** [comment]
@GeorgeS2019 -- this topic may get better visibility if it's moved to some ML.NET forum.

**Comment by GeorgeS2019** [comment]
@tarekgh 
Finally => [Microsoft.ML.Tokenizers](https://github.com/dotnet/machinelearning/pull/6272)

@NiklasGustafsson 
Ideally, we need to open the discussion for the repo [ML.NET](https://github.com/dotnet/machinelearning/)
